File: InteractiveLive/Android/VideoOne/component/ToolKit/src/main/java/com/vertcdemo/core/utils/AppUtil.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.core.utils;

import android.app.Application;

public class AppUtil {

    private static Application mAppCxt;

    public static void initApp(Application app) {
        mAppCxt = app;
    }

    public static Application getApplicationContext() {
        if (mAppCxt == null) {
            throw new IllegalStateException("Please init app at first!");
        } else {
            return mAppCxt;
        }
    }
}

File: Client/Android/component/solution-base/src/main/kotlin/com/vertcdemo/core/utils/AppUtil.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.core.utils

import android.app.Application

object AppUtil {
    private var mAppCxt: Application? = null

    @JvmStatic
    fun initApp(app: Application?) {
        mAppCxt = app
    }

    @JvmStatic
    val applicationContext: Application
        get() = mAppCxt ?: throw IllegalStateException("Please init app at first!")
}
